src/main/java/hudson/maven/MavenProbeAction.java

```diff
@@ -55,7 +55,7 @@
     private final transient Channel channel;
     private final transient AbstractMavenBuild<?,?> build;
 
-    public final AbstractProject<?,?> owner;
+    public final transient AbstractProject<?,?> owner;
```

Comment:
🐛 , see the review comment

Reply:
I do not see what the problem is. It can indeed happen that `Run.save` is called in the middle of the build—in fact this is likely. So a `MavenProbeAction` will temporarily be included in `build.xml`. But it does not matter—the action is removed in a `finally` block by the time the build ends. Even if Jenkins were `SIGKILL`ed and you loaded the `build.xml` after a restart (when it was a `FAILURE`), `owner` would indeed be null, but `MavenProbeAction` would not have actually worked anyway.